test/native/helpers.js

```diff
@@ -113,3 +115,16 @@ export function getEditorHtml() {
 	triggerHtmlSerialization();
 	return serializedHtml;
 }
+
+/**
+ * @deprecated Shallow rendering React component is usually not a good idea, please use `render` instead.
+ *
+ * @param {React.Element} instance The node to render.
+ * @return {Object} The wrapper instance around the rendered output.
+ */
+export function shallow( instance ) {
+	const renderer = new ShallowRenderer();
+	renderer.render( React.createElement( instance.type, instance.props ) );
+
+	return { output: renderer.getRenderOutput() };
+}
```

Comment:
So far, I took the approach of completing the minimal changes required to remove Enzyme. However, I agree we could go further to completely remove `shallow`. That is — or likely should be — our long-term goal, so I am happy to implement that now within this PR. 

I will note that while your proposed solutions are accurate and simple, the majority of the work comes from switching from rendering a _single component_ (shallow) to a _component and its children_ (deep).  However, it is likely still manageable for this PR.

Reply:
> So far, I took the approach of completing the minimal changes required to remove Enzyme. However, I agree we could go further to completely remove shallow. That is — or likely should be — our long-term goal, so I am happy to implement that now within this PR.

I'm happy to keep the `shallow` function in order to prevent getting this PR bigger, so please feel free to skip this suggestion as it could be done in a separate PR 🙇 .

> I will note that while your proposed solutions are accurate and simple, the majority of the work comes from switching from rendering a single component (shallow) to a component and its children (deep). However, it is likely still manageable for this PR.

Good point, the change is not trivial indeed, as deep rendering might produce unexpected side effects on the tests that used shallow rendering. I briefly checked the tests files I listed in https://github.com/WordPress/gutenberg/pull/37715#discussion_r781417210 and none of them looked too complex to make the change, actually, I think you covered the most difficult ones in this PR 💪 .

In any case, as I commented above, we could do this in a separate PR, I wouldn't like to unnecessarily extend this PR due to this.

Reply:
All `shallow` usage was removed in https://github.com/WordPress/gutenberg/pull/37715/commits/a0cb8639b7428a4ca8b756f11d65ee636321b532.